src/dispatch/event/service.py

```diff
@@ -79,6 +81,13 @@ def delete(*, db_session, event_id: int):
     db_session.commit()
 
 
+def log_subject_event(subject: Subject, id: int, **kwargs) -> Event:
+    if isinstance(subject, Incident):
+        return log_incident_event(incident_id=id, **kwargs)
+    else:
+        return log_case_event(case_id=id, **kwargs)
+
+
```

Comment:
```suggestion
def log_subject_event(subject: Subject, **kwargs) -> Event:
    if isinstance(subject, Incident):
        return log_incident_event(incident_id=subject.id, **kwargs)
    else:
        return log_case_event(case_id=subject.id, **kwargs)


```

already passing `subject` so we can use the associated `id`